### PR TITLE
python310Packages.scikit-image: 0.19.3 -> 0.21.0

### PR DESCRIPTION
**Pull request by natsukium**
Diff: https://github.com/scikit-image/scikit-image/compare/v0.19.3...v0.21.0

Changelog: https://github.com/scikit-image/scikit-image/releases/tag/v0.21.0

This PR fixes some build failures related to scipy updates.
For example, `python310Packages.albumentations` will now build correctly.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 249674` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.caffe</li>
    <li>python310Packages.caffe.bin</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
  </ul>
</details>
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li>intensity-normalization (python310Packages.intensity-normalization)</li>
    <li>intensity-normalization.dist (python310Packages.intensity-normalization.dist)</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.spectral-cube</li>
    <li>python310Packages.spectral-cube.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.intensity-normalization</li>
    <li>python311Packages.intensity-normalization.dist</li>
    <li>python311Packages.spectral-cube</li>
    <li>python311Packages.spectral-cube.dist</li>
  </ul>
</details>
<details>
  <summary>88 packages built:</summary>
  <ul>
    <li>easyocr (python310Packages.easyocr)</li>
    <li>easyocr.dist (python310Packages.easyocr.dist)</li>
    <li>napari (python310Packages.napari)</li>
    <li>napari.dist (python310Packages.napari.dist)</li>
    <li>ome_zarr (python310Packages.ome-zarr)</li>
    <li>ome_zarr.dist (python310Packages.ome-zarr.dist)</li>
    <li>python310Packages.albumentations</li>
    <li>python310Packages.albumentations.dist</li>
    <li>python310Packages.aplpy</li>
    <li>python310Packages.aplpy.dist</li>
    <li>python310Packages.batchgenerators</li>
    <li>python310Packages.batchgenerators.dist</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.boxx.dist</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.dask-image</li>
    <li>python310Packages.dask-image.dist</li>
    <li>python310Packages.glymur</li>
    <li>python310Packages.glymur.dist</li>
    <li>python310Packages.imagecorruptions</li>
    <li>python310Packages.imagecorruptions.dist</li>
    <li>python310Packages.insightface</li>
    <li>python310Packages.insightface.dist</li>
    <li>python310Packages.lime</li>
    <li>python310Packages.lime.dist</li>
    <li>python310Packages.mung</li>
    <li>python310Packages.mung.dist</li>
    <li>python310Packages.muscima</li>
    <li>python310Packages.muscima.dist</li>
    <li>python310Packages.omrdatasettools</li>
    <li>python310Packages.omrdatasettools.dist</li>
    <li>python310Packages.pims</li>
    <li>python310Packages.pims.dist</li>
    <li>python310Packages.psd-tools</li>
    <li>python310Packages.psd-tools.dist</li>
    <li>python310Packages.scikit-image</li>
    <li>python310Packages.scikit-image.dist</li>
    <li>python310Packages.slicedimage</li>
    <li>python310Packages.slicedimage.dist</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>python310Packages.zcs</li>
    <li>python310Packages.zcs.dist</li>
    <li>python311Packages.albumentations</li>
    <li>python311Packages.albumentations.dist</li>
    <li>python311Packages.aplpy</li>
    <li>python311Packages.aplpy.dist</li>
    <li>python311Packages.batchgenerators</li>
    <li>python311Packages.batchgenerators.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.caffe</li>
    <li>python311Packages.caffe.bin</li>
    <li>python311Packages.caffeWithCuda</li>
    <li>python311Packages.caffeWithCuda.bin</li>
    <li>python311Packages.dask-image</li>
    <li>python311Packages.dask-image.dist</li>
    <li>python311Packages.easyocr</li>
    <li>python311Packages.easyocr.dist</li>
    <li>python311Packages.glymur</li>
    <li>python311Packages.glymur.dist</li>
    <li>python311Packages.imagecorruptions</li>
    <li>python311Packages.imagecorruptions.dist</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.insightface.dist</li>
    <li>python311Packages.lime</li>
    <li>python311Packages.lime.dist</li>
    <li>python311Packages.mung</li>
    <li>python311Packages.mung.dist</li>
    <li>python311Packages.muscima</li>
    <li>python311Packages.muscima.dist</li>
    <li>python311Packages.ome-zarr</li>
    <li>python311Packages.ome-zarr.dist</li>
    <li>python311Packages.omrdatasettools</li>
    <li>python311Packages.omrdatasettools.dist</li>
    <li>python311Packages.pims</li>
    <li>python311Packages.pims.dist</li>
    <li>python311Packages.psd-tools</li>
    <li>python311Packages.psd-tools.dist</li>
    <li>python311Packages.scikit-image</li>
    <li>python311Packages.scikit-image.dist</li>
    <li>python311Packages.slicedimage</li>
    <li>python311Packages.slicedimage.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/scikit-image/default.nix [hunk 1]
```diff
@@ -3,53 +3,98 @@
...
+      optional = [
+        astropy
+        cloudpickle
+        dask
+        matplotlib
+        pooch
+        pyamg
+        scikit-learn
+        simpleitk
+      ] ++ dask.optoinals-dependencies.array;
```
Comment (yl3dy) [inline pkgs/development/python-modules/scikit-image/default.nix:95]: Isn't it a typo?

**Review by yl3dy — CHANGES_REQUESTED** [review]

**Comment by yl3dy** [comment]
Result of `nixpkgs-review pr 249674` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.caffe</li>
    <li>python310Packages.caffe.bin</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
  </ul>
</details>
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>intensity-normalization (python310Packages.intensity-normalization)</li>
    <li>intensity-normalization.dist (python310Packages.intensity-normalization.dist)</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.spectral-cube</li>
    <li>python310Packages.spectral-cube.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.intensity-normalization</li>
    <li>python311Packages.intensity-normalization.dist</li>
    <li>python311Packages.spectral-cube</li>
    <li>python311Packages.spectral-cube.dist</li>
  </ul>
</details>
<details>
  <summary>90 packages built:</summary>
  <ul>
    <li>easyocr (python310Packages.easyocr)</li>
    <li>easyocr.dist (python310Packages.easyocr.dist)</li>
    <li>napari (python310Packages.napari)</li>
    <li>napari.dist (python310Packages.napari.dist)</li>
    <li>ome_zarr (python310Packages.ome-zarr)</li>
    <li>ome_zarr.dist (python310Packages.ome-zarr.dist)</li>
    <li>python310Packages.albumentations</li>
    <li>python310Packages.albumentations.dist</li>
    <li>python310Packages.aplpy</li>
    <li>python310Packages.aplpy.dist</li>
    <li>python310Packages.batchgenerators</li>
    <li>python310Packages.batchgenerators.dist</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.boxx.dist</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.dask-image</li>
    <li>python310Packages.dask-image.dist</li>
    <li>python310Packages.glymur</li>
    <li>python310Packages.glymur.dist</li>
    <li>python310Packages.imagecorruptions</li>
    <li>python310Packages.imagecorruptions.dist</li>
    <li>python310Packages.insightface</li>
    <li>python310Packages.insightface.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.lime</li>
    <li>python310Packages.lime.dist</li>
    <li>python310Packages.mung</li>
    <li>python310Packages.mung.dist</li>
    <li>python310Packages.muscima</li>
    <li>python310Packages.muscima.dist</li>
    <li>python310Packages.omrdatasettools</li>
    <li>python310Packages.omrdatasettools.dist</li>
    <li>python310Packages.pims</li>
    <li>python310Packages.pims.dist</li>
    <li>python310Packages.psd-tools</li>
    <li>python310Packages.psd-tools.dist</li>
    <li>python310Packages.scikit-image</li>
    <li>python310Packages.scikit-image.dist</li>
    <li>python310Packages.slicedimage</li>
    <li>python310Packages.slicedimage.dist</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>python310Packages.zcs</li>
    <li>python310Packages.zcs.dist</li>
    <li>python311Packages.albumentations</li>
    <li>python311Packages.albumentations.dist</li>
    <li>python311Packages.aplpy</li>
    <li>python311Packages.aplpy.dist</li>
    <li>python311Packages.batchgenerators</li>
    <li>python311Packages.batchgenerators.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.caffe</li>
    <li>python311Packages.caffe.bin</li>
    <li>python311Packages.caffeWithCuda</li>
    <li>python311Packages.caffeWithCuda.bin</li>
    <li>python311Packages.dask-image</li>
    <li>python311Packages.dask-image.dist</li>
    <li>python311Packages.easyocr</li>
    <li>python311Packages.easyocr.dist</li>
    <li>python311Packages.glymur</li>
    <li>python311Packages.glymur.dist</li>
    <li>python311Packages.imagecorruptions</li>
    <li>python311Packages.imagecorruptions.dist</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.insightface.dist</li>
    <li>python311Packages.lime</li>
    <li>python311Packages.lime.dist</li>
    <li>python311Packages.mung</li>
    <li>python311Packages.mung.dist</li>
    <li>python311Packages.muscima</li>
    <li>python311Packages.muscima.dist</li>
    <li>python311Packages.ome-zarr</li>
    <li>python311Packages.ome-zarr.dist</li>
    <li>python311Packages.omrdatasettools</li>
    <li>python311Packages.omrdatasettools.dist</li>
    <li>python311Packages.pims</li>
    <li>python311Packages.pims.dist</li>
    <li>python311Packages.psd-tools</li>
    <li>python311Packages.psd-tools.dist</li>
    <li>python311Packages.scikit-image</li>
    <li>python311Packages.scikit-image.dist</li>
    <li>python311Packages.slicedimage</li>
    <li>python311Packages.slicedimage.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>

**Review by yl3dy — APPROVED** [review]
LGTM, great work!

**Comment by natsukium** [comment]
Thanks for checking.
Some tests on the aarch64-darwin failed, so I reported them upstream and disabled them.
https://github.com/scikit-image/scikit-image/issues/7104

**Pull request merged**